Doc/Manual/Fortran.html

```diff
@@ -0,0 +1,2192 @@
+</p>
+
+
+<p>
+Generally, these typemaps are defined as applying to arguments called <code>NATIVE</code>; they can be applied to <em>all</em> arguments regardless of name with the <code>%apply</code> directive:
+</p>
+
+
+<div class="code"><pre><code>%apply const char* NATIVE { const char* };
+</code></pre>
+</div>
+
+
+<p>
+or to the output of a specific function such as <code>const char* get_foo_string(int i);</code> with
+</p>
+
+
+<div class="code"><pre><code>%apply const char* NATIVE { get_foo_string };
+</code></pre>
+</div>
+
+<H3><a name="Fortran_stdstring">Std::string</a></H3>
+
+<p>
+A special set of typemaps is provided that transparently converts native Fortran character strings to and from <code>std::string</code> classes. It operates essentially like the <a href="#Fortran_byte_strings">byte strings</a> described above: it can transparently convert strings of data, even those with embedded null characters, to and from Fortran. This typemap is provided in <code>&lt;std_string.i&gt;</code>.
+</p>
+
+<H3><a name="Fortran_stdvector">Std::vector</a></H3>
+
+<p>
+The C <code>std::vector</code> class is included with its basic methods. Several typemaps are included alongside it that allow for seamless interoperability with Fortran arrays (with some performance penalty from extra memory allocations and copies). To instantiate wrappers for <code>std::vector&lt;double&gt;</code> as a class named <code>VecDbl</code>, write
+</p>
+
+
+<div class="code"><pre><code>%include &lt;std_vector.i&gt;
+%template(VecDbl) std::vector&lt;double&gt;;
+</code></pre>
+</div>
+
+
+<p>
+If your code does not use <code>std::vector</code> to do any heavy lifting (i.e. your vectors are small and not shared), you can choose to transparently convert those arguments to and from Fortran arrays. This can be done with:
+</p>
+
+
+<div class="code"><pre><code>%include &lt;std_vector.i&gt;
+%template() std::vector&lt;double&gt;;
+%apply const std::vector&lt;double&gt; NATIVE&amp; { const std::vector&lt;double&gt;&amp; }
+%apply std::vector&lt;double&gt; NATIVE { std::vector&lt;double&gt; }
+</code></pre>
+</div>
+
+
+<p>
+to allow
+</p>
+
+<pre class="c++"><code>std::vector&lt;double&gt; get_values();
+</code></pre>
+
+<p>
+to return a Fortran <code>real(C_DOUBLE), dimension(:), allocatable</code> array.
+</p>
+
+<H3><a name="Fortran_other_c_standard_library_containers">Other C++ standard library containers</a></H3>
+
+<p>
+Other useful types such as <code>std::map</code>, <code>std::set</code>, have no or minimal implementation. Contributions to these classes (by changes to <code>swig/Library/fortran/std_{cls}.i</code>) will be warmly welcomed.
+</p>
+
+<H3><a name="Fortran_smart_pointers">Smart pointers</a></H3>
+
```

Comment:
Are you talking about SWIGDocumentation.html#SWIGPlus_smart_pointers ? I haven't tested it (since I thought its use was effectively deprecated by the `%shared_ptr` macro) but will document it.

Reply:
Yes, correct. %shared_ptr only works for one particular smart pointer (std::shared_ptr). There are lots of others and the default handling for these is as documented in this main section. I'm sure it will just work for Fortran as this is functionality provided by the core. Try port any of the `Examples/test-suite/python/smart_pointer*_runme.py` runtime tests to test it out.

Reply:
OK, currently the unary `operator *` and `operator ->` are ignored by default, but I've added [an issue](https://github.com/sethrj/swig/issues/76) to add support.